main/test/ca/uwaterloo/flix/language/phase/TestParser.scala

```diff
@@ -1456,27 +1456,27 @@ class TestParser extends FunSuite with TestUtils {
 
   test("Expression.MatchLambda.05") {
     val input = "def f: Option[Int] -> Int = match None -> 42"
-    run(input, core = false)
+    expectError[NonExhaustiveMatchError](new Flix().addStr(input).compile())
```

Comment:
The MatchLambdas here are non exhaustive, but I wonder if between those and the Let Matches, maybe instead of not letting them be non exhaustive, maybe we don't check them, or compile them to have and implicit _ -> MatchError . It seems to me that if you're doing something like the above, you fully know that it's non exhaustive, but it still seems like an nice feature to have in the case where you know that it can't be a Some(_), but you can't prove it.